contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,333 @@
+        _;
+    }
+
+    /**
+     * @notice Construct the EventBasedPredictionMarket
+     * @param params Constructor params used to initialize the LSP. Key-valued object with the following structure:
+     *    - `pairName`: Name of the long short pair tokens created for the prediction market.
+     *    - `priceIdentifier`: Price identifier, registered in the DVM for the long short pair.
+     *    - `collateralToken`: Collateral token used to back LSP synthetics.
+     *    - `financialProductLibrary`: Contract providing settlement payout logic.
+     *    - `customAncillaryData`: Custom ancillary data to be passed along with the price request to the OO.
+     *    - `finder`: DVM finder to find other UMA ecosystem contracts.
+     *    - `timerAddress`: Timer used to synchronize contract time in testing. Set to 0x000... in production.
+     */
+    constructor(ConstructorParams memory params) Testable(params.timerAddress) {
+        expirationTimestamp = getCurrentTime(); // Set the request timestamp to the current block timestamp.
+
+        // Holding long tokens gives the owner exposure to the long position,
+        // i.e. the case where the answer to the prediction market question is YES.
+        longToken = new ExpandedERC20(string(abi.encodePacked(params.pairName, " Long Token")), "PLT", 18);
+        // Holding short tokens gives the owner exposure to the short position,
+        // i.e. the case where the answer to the prediction market question is NO.
+        shortToken = new ExpandedERC20(string(abi.encodePacked(params.pairName, " Short Token")), "PST", 18);
+
+        // Add burner and minter required roles to the long and short tokens.
+        longToken.addMinter(address(this));
+        shortToken.addMinter(address(this));
+        longToken.addBurner(address(this));
+        shortToken.addBurner(address(this));
+
+        finder = params.finder;
+
+        collateralToken = params.collateralToken;
+        financialProductLibrary = params.financialProductLibrary;
+        customAncillaryData = params.customAncillaryData;
+        priceIdentifier = params.priceIdentifier;
+        pairName = params.pairName;
+    }
+
+    // Requests the price from the optimistic oracle
+    // The caller must have sufficient balance to pay the proposer reward and approve the contract to spend the collateral.
+    function initializeMarket() public {
+        // If the proposer reward was set then pull it from the caller of the function.
+        if (proposerReward > 0) {
+            collateralToken.safeTransferFrom(msg.sender, address(this), proposerReward);
+        }
+        _requestOraclePrice();
+    }
+
+    // Request a price in the optimistic oracle for a given request timestamp and ancillary data combo. Set the bonds
+    // accordingly to the deployer's parameters. Will revert if re-requesting for a previously requested combo.
+    function _requestOraclePrice() internal {
```

Comment:
nit: can you group functions by how they are exposed? i.e put all internal functions together at the bottom.